lib/client/ipam_block_reader_writer.go

```diff
@@ -82,10 +82,9 @@ func (rw blockReaderWriter) claimNewAffineBlock(
 		for _, pool := range givenPools {
 			// Validate the given pool is actually configured and matches the version.
 			if err := rw.checkPoolAvailability(pool); err != nil {
-				return nil, err
+				log.Errorf("IPPool not available: %s", err.Error())
 			} else if version.Number != pool.Version() {
-				estr := fmt.Sprintf("The given pool (%s) does not match IP version %d", pool.String(), version.Number)
-				return nil, goerrors.New(estr)
+				log.Errorf("The given pool (%s) does not match IP version %d", pool.String(), version.Number)
 			}
 			pools = append(pools, pool)
```

Comment:
Either this needs to be in an `else` branch, or you need to `continue` in the previous if clauses.